composer/core/state.py

```diff
@@ -272,6 +272,7 @@ def __init__(
 
         # Set defaults for transient variables (to make pyright happy)
         self.batch: Any = None
+        self.deepspeed_config: Any = None
```

Comment:
Can you add this argument (and a docstring entry) as part of the `__init__` for state, and type it as `Optional[Dict[str, Any]]`?